File: app/src/main/java/net/devdome/paperplayer/ui/fragment/PlaylistFragment.java
package net.devdome.paperplayer.ui.fragment;

import android.app.Dialog;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetBehavior;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.design.widget.CoordinatorLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.View;

import net.devdome.paperplayer.R;
import net.devdome.paperplayer.adapter.PlaylistAdapter;
import net.devdome.paperplayer.playback.QueueManager;

/**
 * Created by Michael on 7/4/2016.
 */

public class PlaylistFragment extends BottomSheetDialogFragment {

    private View contentView;

    private BottomSheetBehavior.BottomSheetCallback mBottomSheetBehaviorCallback = new BottomSheetBehavior.BottomSheetCallback() {

        @Override
        public void onStateChanged(@NonNull View bottomSheet, int newState) {
            if (newState == BottomSheetBehavior.STATE_COLLAPSED) {
                dismiss();
            }
        }

        @Override
        public void onSlide(@NonNull View bottomSheet, float slideOffset) {
        }
    };

    @Override
    public void setupDialog(Dialog dialog, int style) {
        super.setupDialog(dialog, style);
        contentView = View.inflate(getActivity(), R.layout.bottomsheet_playlist, null);
        dialog.setContentView(contentView);

        CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) ((View) contentView.getParent()).getLayoutParams();
        final CoordinatorLayout.Behavior behavior = params.getBehavior();

        if (behavior != null && behavior instanceof BottomSheetBehavior) {
            ((BottomSheetBehavior) behavior).setState(BottomSheetBehavior.STATE_EXPANDED);
            ((BottomSheetBehavior) behavior).setBottomSheetCallback(mBottomSheetBehaviorCallback);

        }

        Toolbar toolbar = (Toolbar) contentView.findViewById(R.id.toolbar);
        toolbar.setTitle(getString(R.string.playlist));
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });
        setUpPlaylist();
    }

    private void setUpPlaylist() {
        RecyclerView rvPlaylist = (RecyclerView) contentView.findViewById(R.id.rv_playlist);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
        int currentPlayingIndex = QueueManager.getInstance().getCurrentIndex();
        layoutManager.scrollToPositionWithOffset(currentPlayingIndex, 20);
        layoutManager.supportsPredictiveItemAnimations();
        rvPlaylist.setLayoutManager(layoutManager);
        rvPlaylist.setAdapter(new PlaylistAdapter());
    }
}

File: app/src/main/java/xyz/michaelobi/paperplayer/presentation/player/PlaylistFragment.kt
package xyz.michaelobi.paperplayer.presentation.player

import android.app.Dialog
import android.support.design.widget.BottomSheetBehavior
import android.support.design.widget.BottomSheetDialogFragment
import android.support.design.widget.CoordinatorLayout
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.View
import xyz.michaelobi.paperplayer.R
import xyz.michaelobi.paperplayer.injection.Injector
import xyz.michaelobi.paperplayer.playback.queue.QueueManager


/**
 * PaperPlayer
 * Michael Obi
 * 22 05 2017 1:04 AM
 */
class PlaylistFragment : BottomSheetDialogFragment() {
    val queueManager: QueueManager = Injector.provideQueueManager()


    override fun setupDialog(dialog: Dialog?, style: Int) {
        super.setupDialog(dialog, style)
        val view = View.inflate(activity, R.layout.bottomsheet_playlist, null)
        dialog?.setContentView(view)
        val params = (view.parent as View).layoutParams as CoordinatorLayout.LayoutParams
        val behavior = params.behavior
        if (behavior != null) {
            (behavior as BottomSheetBehavior<*>).state = BottomSheetBehavior.STATE_EXPANDED
            behavior.setBottomSheetCallback(bottomSheetBehaviorCallback)
        }
        val toolbar = view.findViewById(R.id.toolbar) as Toolbar
        toolbar.title = getString(R.string.playlist)
        toolbar.setNavigationOnClickListener({ dismiss() })
        setUpPlaylist(view)
    }

    private fun setUpPlaylist(view: View) {
        val rvPlaylist = view.findViewById(R.id.rv_playlist) as RecyclerView
        val layoutManager = LinearLayoutManager(activity)
        val currentPlayingIndex = queueManager.currentIndex
        layoutManager.scrollToPositionWithOffset(currentPlayingIndex, 20)
        layoutManager.supportsPredictiveItemAnimations()
        rvPlaylist.layoutManager = layoutManager
        rvPlaylist.adapter = PlaylistAdapter()
    }


    private val bottomSheetBehaviorCallback = object : BottomSheetBehavior.BottomSheetCallback() {
        override fun onStateChanged(bottomSheet: View, newState: Int) {
            if (newState == BottomSheetBehavior.STATE_COLLAPSED) {
                dismiss()
            }
        }

        override fun onSlide(bottomSheet: View, slideOffset: Float) {}
    }
}